app/services/katello/pulp/repository/yum.rb

```diff
@@ -75,19 +75,33 @@ def regenerate_applicability
           smart_proxy.pulp_api.extensions.repository.regenerate_applicability_by_ids([repo.pulp_id], true)
         end
 
+        def generate_mapping(destination_repo)
+          source_repo_map = {}
+          repo.siblings.each do |sibling|
+            key = sibling.library_instance? ? sibling.id : sibling.library_instance_id
+            source_repo_map[key] = sibling.pulp_id
+          end
+
+          Hash[destination_repo.siblings.map { |sibling| [source_repo_map[sibling.library_instance_id], sibling.pulp_id] }]
+        end
+
         def copy_contents(destination_repo, options = {})
-          override_config = Katello::Repository.build_override_config(options)
+          override_config = Katello::Repository.build_override_config(
+                                                options.merge(repository_mapping: generate_mapping(destination_repo)))
           rpm_copy_clauses, rpm_remove_clauses = generate_copy_clauses(options[:filters]&.yum(false), options[:rpm_filenames])
-          tasks = [smart_proxy.pulp_api.extensions.rpm.copy(repo.pulp_id, destination_repo.pulp_id,
-                   rpm_copy_clauses.merge(:override_config => override_config))]
+          tasks = []
+          if rpm_copy_clauses
+            tasks << smart_proxy.pulp_api.extensions.rpm.copy(repo.pulp_id, destination_repo.pulp_id,
```

Comment:
kind of a nitpick, but can you move the 'override_config' in this if statement?  That way its more clear its only used for rpm copying

Reply:
to clarify, move the definition of override_config into this if statement

Reply:
thoughts on this?

Reply:
I agree with this too

Reply:
updated